src/osd/PGLog.h

```diff
@@ -78,7 +78,11 @@ struct PGLog : DoutPrefixProvider {
     mutable ceph::unordered_multimap<osd_reqid_t,pg_log_entry_t*> extra_caller_ops;
     mutable ceph::unordered_map<osd_reqid_t,pg_log_dup_t*> dup_index;
 
-    // recovery pointers
+    // recovery pointers.  the *complete_to* pointer is used for updating
+    // last_complete incrementally during the course of recovery.  It is not
+    // strictly required to be correct, as the last_complete value will also
```

Comment:
I don't like this comment either. We update the last_complete field based on the complete_to pointer during recovery. If the complete_to pointer is not correct, then we have problems.

It would be more appropriate to saying `the complete_to pointer might become invalid during recovery, e.g., due to the owner pg currently having no log entries in the list. `. I don't think it is necessary to mention, though...